2023-04-21 - Futher recommended changes

Krelsis opened the pull request:
# Summary

- Updated dependency versions.
- Added typescript definitions for typescript support.
    - Now copied into dist via rollup.
- Added consideration for initial state/external modification - provided values on mount no longer reset to default values.
    - useState watching values for changes and invokes setDaysOfWeek respectively.
- Added ability to customise the render order using a new parameter
    - Will re-order the display of the days without having to change their position in the days array.
    - Some libraries/data types will treat sunday as 0 rather than 6, to circumvent the need to rectify the data in post, we can now just change the order in which they are renderred.
    ```javascript
		<WeekdaysInput
			days={[
				'sunday',
				'monday',
				'tuesday',
				'wednesday',
				'thursday',
				'friday',
				'saturday'
			]} // sunday, monday, tuesday, wednesday, thursday, friday, saturday, sunday
			renderOrder={[1, 2, 3, 4, 5, 6, 0]} // render the 0th day in the 7th position.
			showChars={3} // mon, tue, wed, thu, fri, sat, sun
			textCase={"firstToUpper"} // Mon, Tue, Wed, Thu, Fri, Sat, Sun
		/>
    ```
- Added ability to customise the wrapper of the days selection input as well as the selector themselves with 2 new classes.
- Added overload for onChange function which will now return an array of numbers if the provided value was an array of numbers.
    - Circumvents the need for sorting the string into an array according to you (outer-component) state.
    - Potential breaking change if you already handle the above.
- Updated readme.
- Updated version number.
- A fair bit of reformatting, sorry about that.

axelmy318 commented:
Hi, thanks alot for the PR! The changes look great however I'm having issues testing it.

In the WeekdaysInput story, I print the `daysAsObject` and `days` states but they don't update when I use the input...
Does it work on your end ?

Krelsis commented:
> Hi, thanks alot for the PR! The changes look great however I'm having issues testing it.
> 
> In the WeekdaysInput story, I print the `daysAsObject` and `days` states but they don't update when I use the input... Does it work on your end ?

Will have to bare with me a moment, having issues building storybook, `'build-storybook' is not recognized as an internal or external command,operable program or batch file.` is being thrown, I must be missing a local dependency.
^ I've only tested from my app where I have this imported.

axelmy318 commented:
Just type `npx storybook@next automigrate` and then say yes to everything it asks. After that you can change start-storybook to `sb dev` in the package.json and run `npm run storybook`

Krelsis commented:
I've been doing some playing around, so far, it works perfectly on my app, on storybook, it doesn't work, seems that although the component calls onChange() the storybook function just doesn't get called. 
Will do some digging to see if there is a quick fix.

axelmy318 commented:
Great. Keep me posting, thanks

Krelsis commented:
Just an assumption, but I believe storybook didn't quite like the Proptypes.func definition for onChange.
I've changed storybook to work with typescript and updated the story to be a ts version, and now it works properly on my end. 

Go ahead and try it out :) 

Considering the plain JS version of storybook didn't quite agree with it, I'd like to make sure that non TS projects can actually use this properly.

I don't have any plain JS (non ts) projects at hand that I can add this package to.

Do you have any plain JS projects you can test this branch on before merging?
Would be a simple rollup and copy/paste `package.json` + `dist` into the `node_modules/react-weekdays-input` folder of your target project. And test that your outer state gets updated.

axelmy318 commented:
Okay. I'll test it today on a non ts project

axelmy318 commented:
I was able to test is on a plain JS project and it worked fine! 
Merged + published 1.3.0
Thank you for the changes :)

Krelsis commented:
Thank you for your swift responses :)
